homeassistant/components/hydrawise/__init__.py

```diff
@@ -26,37 +35,63 @@
     extra=vol.ALLOW_EXTRA,
 )
 
+PLATFORMS: list[Platform] = [Platform.BINARY_SENSOR, Platform.SENSOR, Platform.SWITCH]
+
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Hunter Hydrawise component."""
-    conf = config[DOMAIN]
-    access_token = conf[CONF_ACCESS_TOKEN]
-    scan_interval = conf.get(CONF_SCAN_INTERVAL)
+    if DOMAIN not in config:
+        return True
 
+    async_create_issue(
```

Comment:
This will not be enough. See https://github.com/home-assistant/architecture/blob/master/adr/0021-YAML-integration-configuration-deprecation-policy.md

See #97673 as a small example on what might be needed.

Reply:
Okay, I followed a similar model to your example. I think that addresses everything in the ADR.